documentation/testing.md

```diff
@@ -0,0 +1,79 @@
+Local Testing
+==========================
+
+For local testing Streisand includes a `Vagrantfile` that creates two virtual
+machines: `streisand-host` and `streisand-client`.
+
+The `streisand-host` machine is provisioned with the standard Streisand
+playbooks and replicates a Streisand server created with a cloud provider, but
+running on your local computer.
+
+The `streisand-client` machine is provisioned specifically to act as a client of
+the `streisand-host`. It connects to the `streiand-host`'s HTTPS gateway to
+download client configuration files that are used by test scripts to ensure that
+services work "end-to-end".
+
+Using Vagrant for Local Testing
+-------------------------------
+
+1. [Install Vagrant](https://www.vagrantup.com/docs/installation/)
+2. Clone the Streisand repository and enter the directory.
+
+       git clone https://github.com/jlund/streisand.git && cd streisand
+3. If this is your first time following these steps, create & start the
+   `streisand-host` and `streisand-client` virtual machines with:
+
+       `vagrant up`
+4. If you have already created the virtual machines and wish to re-run the
+   Streisand playbooks then re-provision the virtual machines with:
+
```

Comment:
I feel this could be better re-worded to "...to re-run the Streisand playbooks, the virtual machines can be re-provisioned with:"